Vectra/vectra_modules/connector_vectra_entity_scoring.py

```diff
@@ -0,0 +1,277 @@
+
+    def log_exception(self, *args, **kwargs) -> None:
+        self.connector.log_exception(*args, **kwargs)
+
+    @property
+    def running(self) -> bool:
+        return not self._stop_event.is_set()
+
+    def stop(self) -> None:
+        self._stop_event.set()
+
+    @property
+    def stepper(self) -> TimeStepper:
+        with self.connector.context as cache, self.connector.context_lock:
+            most_recent_date_requested_str = cache.get(self.entity_type, {}).get("last_event_date")
+
+        if most_recent_date_requested_str is None:
+            return TimeStepper.create(
+                self,
+                self.configuration.frequency,
+                self.configuration.timedelta,
+                self.configuration.start_time,
+            )
+
+        # parse the most recent requested date
+        most_recent_date_requested = isoparse(most_recent_date_requested_str)
+
+        now = datetime.now(timezone.utc)
+        one_month_ago = now - timedelta(days=30)
+        if most_recent_date_requested < one_month_ago:
+            most_recent_date_requested = one_month_ago
+
+        return TimeStepper.create_from_time(
+            self,
+            most_recent_date_requested,
+            self.configuration.frequency,
+            self.configuration.timedelta,
+        )
+
+    def update_stepper(self, new_datetime: datetime) -> None:
+        with self.connector.context as cache, self.connector.context_lock:
+            if self.entity_type not in cache:
+                cache[self.entity_type] = {}
+
+            cache[self.entity_type]["last_event_date"] = new_datetime.isoformat()
+
+    def load_events_cache(self) -> Cache:
+        cache: Cache = LRUCache(maxsize=self.cache_size)
+
+        with self.connector.context_lock:
+            with self.connector.cache_context as context:
+                # load the cache from the context
+                events_cache = context.get(self.entity_type, {}).get("events_cache", [])
+
+        for event_hash in events_cache:
+            cache[event_hash] = True
+
+        return cache
+
+    def update_cache(self):
+        with self.connector.cache_context as cache, self.connector.context_lock:
+            if self.entity_type not in cache:
+                cache[self.entity_type] = {}
+
+            cache[self.entity_type]["events_cache"] = list(self.events_cache.keys())
+
+    def filter_processed_events(self, events: list[dict]) -> Generator[dict, None, None]:
+        for event in events:
+            event_id = event["id"]
+            if event_id in self.events_cache:
+                continue
+
+            self.events_cache[event_id] = True
+            yield event
+
+    def __get_request(self, *args, **kwargs) -> Response:
+        try:
+            response = self.client.get(*args, **kwargs)
+            response.raise_for_status()
+
+            return response
+
+        except requests.exceptions.HTTPError as err:
+            self.log_exception(err)
```

Comment:
I could be interesting to extract the payload of the response and log it as we (and the customer) can get information about the reason of the error.